### PR TITLE
Better git integration + Real time updates!

### PR DESCRIPTION
**Pull request by PatrickShaw**
## Proposed changes
- Pushing wasn't working with GitHub since shallow fetches weren't working so that's been disabled.
- OpenAPI Platform wasn't actually staging deleted files so that's been fixed.
- The build status updates in real time. First time we're actually making use of the real-time API that featherjs provides which is pretty cool.

## Types of changes

What types of changes does your code introduce to OpenAPI Platform?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I have signed and submitted the appropriate [CLA](../docs/contrib)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/frontend-dist/src/view/basic/SdkConfigItem.tsx [hunk 1]
```diff
@@ -48,14 +48,23 @@ export class SdkConfigItem extends Component<SdkConfigItemProps> {
...
+    // TODO: Little hacky changing modifying one's own props
+    this.props.sdkConfig.buildStatus = BuildStatus.Building;
+    client
       .service('sdks')
-      .create({ sdkConfigId: this.props.sdkConfig.id });
-    // TODO: Need to get this from the actual backend
-    this.props.sdkConfig.buildStatus = BuildStatus.Success;
-    this.latestSdkUrl = sdk.path;
+      .create({ sdkConfigId: this.props.sdkConfig.id })
+      .then(
```
Comment (calebj0seph) [inline packages/frontend-dist/src/view/basic/SdkConfigItem.tsx]: This is an async function, why not sue await here?

File: packages/frontend-dist/src/state/SdkConfigState.tsx [hunk 2]
```diff
@@ -1,11 +1,62 @@
...
+        $sort: {
+          createdAt: 1,
+        },
+      },
+    });
+    sdkConfigs.forEach(sdkConfig => {
+      state.sdkConfigs.set(sdkConfig.id, sdkConfig);
+    });
+  }
+
```
Comment (calebj0seph) [inline packages/frontend-dist/src/state/SdkConfigState.tsx]: Would it be hard to add realtime updating for specs too?

File: packages/frontend-dist/src/view/basic/SdkConfigItem.tsx [hunk 3]
```diff
@@ -48,14 +48,23 @@ export class SdkConfigItem extends Component<SdkConfigItemProps> {
...
+  /**
+   * TODO: Not really sure when this got approved but this method really shouldn't be inside of a
+   * basic component.
+   */
   @action
   public createSdk = async () => {
-    const sdk: HasId<Sdk> = await client
+    this.latestSdkUrl = undefined;
+    // TODO: Little hacky changing modifying one's own props
+    this.props.sdkConfig.buildStatus = BuildStatus.Building;
```
[the remark this answers is not in the document]
Comment (PatrickShaw) [reply packages/frontend-dist/src/view/basic/SdkConfigItem.tsx]: Yeah but it takes time for server to respond with a build status change. May as well do it client side as well so it's instant.

File: packages/frontend-dist/src/state/SdkConfigState.tsx (hunk 2 above)
Comment (PatrickShaw) [reply packages/frontend-dist/src/state/SdkConfigState.tsx]: Later :)

File: packages/frontend-dist/src/view/basic/SdkConfigItem.tsx (hunk 1 above)
Comment (PatrickShaw) [reply packages/frontend-dist/src/view/basic/SdkConfigItem.tsx]: I think I originally had 2 async calls inside of this and I wanted this one to happen without waiting for a response. That's not the case now so i've changed it back 👍

**Pull request merged**